File: cases/taint/java/return-relay-one-hop-negative/ReturnRelayOneHopNegative.java
package dataflowbench.taint;

final class ReturnRelayOneHopNegative {
    static int dfb_source() { // DFB-SOURCE: return-one-hop-negative-input
        return 1;
    }

    static int relay(int value) { // DFB-WITNESS: return-one-hop-negative-relay
        return value;
    }

    static void dfb_sink(int value) { } // DFB-SINK: return-one-hop-negative-sink

    static void run() {
        int result = relay(dfb_source());
        dfb_sink(0);
    }
}

File: cases/taint/csharp/return-relay-one-hop-negative/ReturnRelayOneHopNegative.cs
namespace DataFlowBench;

static class ReturnRelayOneHopNegative
{
    static int dfb_source() // DFB-SOURCE: return-one-hop-negative-input
    {
        return 1;
    }

    static int Relay(int value) // DFB-WITNESS: return-one-hop-negative-relay
    {
        return value;
    }

    static void dfb_sink(int value) { } // DFB-SINK: return-one-hop-negative-sink

    static void Run()
    {
        int result = Relay(dfb_source());
        dfb_sink(0);
    }
}
